doc/userguide/configuration/suricata-yaml.rst

```diff
@@ -2010,19 +2010,17 @@ The NUMA node to which the NIC is connected to can be determined from the file `
     ## cat /sys/class/net/<KERNEL NAME OF THE NIC>/device/numa_node e.g.
     cat /sys/class/net/eth1/device/numa_node
 
-If Suricata has enabled at least 2 (or more) workers, the incoming traffic is load balanced across the worker threads
-by Receive Side Scaling (RSS). Internally, DPDK runmode uses
-a `symmetric hash (0x6d5a) <https://www.ran-lifshitz.com/2014/08/28/symmetric-rss-receive-side-scaling/>`_
+Suricata operates in workers runmode. Packet distribution relies on Receive Side Scaling (RSS), which distributes packets across the NIC queues. 
+Individual Suricata workers then poll packets from the NIC queues. 
+Internally, DPDK runmode uses a `symmetric hash (0x6d5a) <https://www.ran-lifshitz.com/2014/08/28/symmetric-rss-receive-side-scaling/>`_
 that redirects bi-flows to specific workers.
 
-Before Suricata can be run, it is required to allocate sufficient number of hugepages. Suricata allocates continuous block of memory.
-For efficiency, CPU allocates memory in RAM in chunks. These chunks are usually in size of 4096 bytes. DPDK and other memory intensive applications makes use of hugepages.
-Hugepages start at the size of 2MB but they can be as large as 1GB. Lower count of pages (memory chunks) allows faster lookup of page entries.
-The hugepages need to be allocated on the NUMA node where the NIC and CPU resides.
-Otherwise, if the hugepages are allocated only on NUMA node 0 and the NIC is connected to NUMA node 1, then the application will fail to start.
-Therefore, it is recommended to first find out to which NUMA node the NIC is connected to and only then allocate hugepages and set CPU cores affinity to the given NUMA node.
-If the Suricata deployment is using multiple NICs on different NUMA nodes then hugepages must be allocated on all of those NUMA nodes.
-
+Before Suricata can be run, it is required to allocate a sufficient number of hugepages. 
+For efficiency, hugepages are continuous chunks of memory (pages) that are larger (2 MB+) than what is typically used in the operating systems (4 KB).
+A lower count of pages allows faster lookup of page entries. The hugepages need to be allocated on the NUMA node where the NIC and affiniated CPU cores reside.
+For example, if the hugepages are allocated only on NUMA node 0 and the NIC is connected to NUMA node 1, then the application will fail to start.
+As a result, it is advised to identify the NUMA node to which the NIC is attached before allocating hugepages and setting CPU core affinity to that node.
+In case Suricata deployment uses multiple NICs, hugepages must be allocated on each of the NUMA nodes used by the Suricata deployment.
```

Comment:
below this it still suggests 8k 2M pages, which I think is too many. The default config will only use a few hundred in my testing, yet 16G is reserved.

Reply:
Below you have a calculation for how much memory a config requires, I wonder if we could read the available huge pages and then calculate if the config will fit? Then we could give a nice error/warning

Reply:
wrt 8k 2M pages - wanted to change that as well but forgot about it. Thanks.

wrt hugepage calculation - I can look into it. I'll create a ticket.

Reply:
https://redmine.openinfosecfoundation.org/issues/5966